### ISSUE TITLE
Be able to work with MeSH ontology

### ISSUE THREAD

**Issue by kwalcock** [clulab/eidos#416]
I finally notice that there is a MeSH ontology in one of the branches.  Is this really for Eidos and not REACH?  It has some 58,742 nodes with a relatively deep structure and with relatively long names.  It might be worth not flattening out the paths to each of the nodes into strings and instead making use of the tree structure.  I volunteer.

**Comment by MihaiSurdeanu** [comment]
Nice!

So, where is that overhead that Zheng mentioned coming from?

**Comment by kwalcock** [comment]
However, it does take a lot of time to load the file in the first place.  This is without caching at all.

![image](https://user-images.githubusercontent.com/8679738/44936615-51f45980-ad2a-11e8-8370-6704a219f6f0.png)

**Comment by kwalcock** [comment]
I'm checking to see if an older version of processors was used.  There was an issue with the caching of the hash values that was optimized shortly before the hackathon.  I'm not sure how long it took to propagate.

**Comment by MihaiSurdeanu** [comment]
1. We should definitely compress the nodes on the path, as you suggested.
2. I am not sure I understand the statement "We are not using that last part, in this case Calcimycin, for grounding at all". We should be using it...

**Comment by bsharpataz** [comment]
@kwalcock are you also having this convo with @marcovzla ?

**Comment by kwalcock** [comment]
2 is probably a moot point for this MeSH ontology.  We ground against words in the descriptions and examples, but not the name of the node.  However, the ontology has the node name field always duplicated as the first example, so it will be used.  Example:
```
                      examples:
                      - Malonyl Coenzyme A
                      - Malonyl CoA
                      - CoA, Malonyl
                      - Coenzyme A, Malonyl
                      name: Malonyl Coenzyme A
```

On the other hand, the UN ontology does not repeat the name in the examples.  In a case like the one below, perhaps "forestry" should be included in the values to ground against.
```
      - OntologyNode:
        examples:
        - afforestation
        - agroforestry
        - forest management
        - reforestation
        name: forestry
        polarity: 1.0
```

**Comment by kwalcock** [comment]
FWIW the vectors in embeddings_november_2016.txt are not normalized.  Eidos expects them to be and only renormalizes after vector addition.  Previous spot checks of vectors.txt and glove.840B.300d.vectors.txt showed normalized values.  Does @ZhengTang1120 know the background on this?  Shall I normalize or is it better to ask the source for an update?

**Comment by bsharpataz** [comment]
I suspect they aren't normalized on purpose, and so we should move towards
varying grounding approaches for grounding to different ontologies... we
likely need to handle MeSH differently anyway, and seems like that would
include the normalization

**Comment by kwalcock** [comment]
This should be addressed with PR #431
